dpcpp/base/onemkl_bindings.hpp

```diff
@@ -91,36 +94,34 @@ template <>
 struct is_supported<std::complex<double>> : std::true_type {};
 
 
-#define GKO_BIND_DOT(ValueType, Name, Func)                                  \
-    void Name(::cl::sycl::queue &exec_queue, std::int64_t n,                 \
-              const ValueType *x, std::int64_t incx, const ValueType *y,     \
-              std::int64_t incy, ValueType *result)                          \
-    {                                                                        \
-        Func(exec_queue, n, x, incx, y, incy, result);                       \
-    }                                                                        \
-    static_assert(true,                                                      \
-                  "This assert is used to counter the false positive extra " \
-                  "semi-colon warnings")
-
-// Bind the dot for x^T * y
-GKO_BIND_DOT(float, dot, oneapi::mkl::blas::row_major::dot);
-GKO_BIND_DOT(double, dot, oneapi::mkl::blas::row_major::dot);
-GKO_BIND_DOT(std::complex<float>, dot, oneapi::mkl::blas::row_major::dotu);
-GKO_BIND_DOT(std::complex<double>, dot, oneapi::mkl::blas::row_major::dotu);
+#define GKO_DECLARE_DOT(ValueType, Name)                                 \
+    void Name(::cl::sycl::queue &exec_queue, std::int64_t n,             \
+              const ValueType *x, std::int64_t incx, const ValueType *y, \
+              std::int64_t incy, ValueType *result)
```

Comment:
When I use it like cuda, compiler will complain they have multiple.
I use anonymous neamspace originally such that it is built for each file contains onemkl_binding header.

Reply:
have you tried making them `inline`? That's also the reason why most functions in our cuBLAS wrappers are templates, since they are implicitly `inline`/weak symbols

Reply:
Thanks! it works, and I miss inline